olp-cpp-sdk-core/include/olp/core/client/ApiResponse.h

```diff
@@ -38,20 +58,85 @@ namespace client {
  * @tparam Result The result type.
  * @tparam Error The error type.
  */
-template <typename Result, typename Error>
-class ApiResponse {
+template <typename Result, typename Error, typename Payload = void>
+class ApiResponse : public ResponseExtension<Payload> {
  public:
   /**
    * @brief The type of result.
    */
   using ResultType = Result;
+
   /**
    * @brief The type of error.
    */
   using ErrorType = Error;
 
+  /**
+   * @brief The type of additional payload.
+   */
+  using PayloadType = Payload;
+
   ApiResponse() = default;
 
+  /**
+   * @brief Creates the `ApiResponse` instance from a similar response type
+   * without payload. The payload is default initialized.
+   *
+   * @note Enabled only for the use cases when the input response type has no
+   * payload.
+   *
+   * Used for moving the successfully executed request.
+   *
+   * @param other The `ApiResponse` instance.
+   */
+  template <typename P = Payload, typename = typename std::enable_if<
+                                      !std::is_same<P, void>::value>::type>
+  ApiResponse(const ApiResponse<Result, Error, void>& other)  // NOLINT
+      : ResponseExtension<P>(),
+        result_(other.GetResult()),
+        error_(other.GetError()),
+        success_(other.IsSuccessful()) {}
```

Comment:
Why not access directly the members of `other`?
Means:
```suggestion
        result_(other.result_),
        error_(other.error_),
        success_(other.success_) {}
```

Same below.

Reply:
It's not possible when the payload is different.